src/irmin-pack/inode.ml

```diff
@@ -259,27 +255,116 @@ struct
       |> sealr
   end
 
+  (** [Val_impl] defines the structure logic of inodes.
+
+      {3 Inode layout}
+
+      The layout ['a layout] associated to an inode ['a t] defines certain
+      properties of the data structure:
+
+      - When [Total], the inode is self contained and immutable.
+      - When [Partial], chunks of the inode might be missing but they can be
+        fetched from the backend when needed. Mutable pointers act as cache.
+      - When [Truncated], chunks of the inode might be missing. Those chunks are
+        unreachable because the pointer to the backend is missing. The inode is
+        immutable.
+
+      The layout is of an inode is determined from the module [Val], it depends
+      on the way the inode was constructed:
+
+      - When [Total], it originates from [Val.v] or [Val.empty].
+      - When [Partial], it originates from [Val.of_bin], which is only used by
+        [Inode.find].
```

Comment:
it is also used by add and remove, no?

Reply:
both `add` and `remove` accept all kinds of layouts